app/controllers/api/v1/tags_controller.rb

```diff
@@ -0,0 +1,61 @@
+# frozen_string_literal: true
+
+module API
+  module V1
+    # Controller class that is responsible for handling tag requests.
+    class TagsController < ActionController::API
+      include DeviseTokenAuth::Concerns::SetUserByToken
+      include Pundit
+
+      before_action :authenticate_user!
+      before_action :set_tag, only: %i[show posts]
+      before_action :authorize_tag!
+      after_action :verify_authorized
+
+      # GET /tags or /tags.json
+      def index
+        @tag = Tag.all
```

Comment:
Here gonna be N+1 because you serialize:
1) post associated with a tag
2) profiles associated with a tag